neutromeratio/ani.py

```diff
@@ -202,40 +207,45 @@ def __init__(self, alchemical_atoms=[]):
         super().__init__(alchemical_atoms)
 
 
+class Ensemble(torch.nn.ModuleList):
+    """Compute the average output of an ensemble of modules."""
+
+    def forward(self, species_input):
+        # type: (Tuple[torch.Tensor, torch.Tensor]) -> Tuple[torch.Tensor, torch.Tensor]
+        outputs = [x(species_input)[1].double() for x in self]
+        species, _ = species_input
+        energy = sum(outputs) / len(outputs)
+        return species, energy 
+      
+
+def load_model_ensemble(species, prefix, count):
+    """Returns an instance of :class:`torchani.Ensemble` loaded from
+    NeuroChem's network directories beginning with the given prefix.
+    Arguments:
+        species (:class:`collections.abc.Sequence`): Sequence of strings for
+            chemical symbols of each supported atom type in correct order.
+        prefix (str): Prefix of paths of directory that networks configurations
+            are stored.
+        count (int): Number of models in the ensemble.
+    """
+    models = []
+    for i in range(count):
+        network_dir = os.path.join('{}{}'.format(prefix, i), 'networks')
+        models.append(torchani.neurochem.load_model(species, network_dir))
+    return Ensemble(models)
 
-class DoubleAniModel(torchani.nn.ANIModel):
-
-    def forward(self, species_aev):
-        # change dtype
-        species, aev = species_aev
-        species_ = species.flatten()
-        present_species = torchani.utils.present_species(species)
-        aev = aev.flatten(0, 1)
-
-        output = torch.full_like(species_, self.padding_fill,
-                                dtype=torch.float32)
-        
-        for i in present_species:
-            mask = (species_ == i)
-            input_ = aev.index_select(0, mask.nonzero().squeeze())
-            output.masked_scatter_(mask, self[i](input_).squeeze())
-        
-        output = output.view_as(species)
-        return species, self.reducer(output.double(), dim=1)
```

Comment:
Does this do the same thing as https://github.com/choderalab/neutromeratio/pull/5/files?short_path=6b0922c#diff-d7c9242dc9cc2a996c36d417ec3894d1R215-R217 ? I'm not sure